compiler/rustc_hir/src/definitions.rs

```diff
@@ -424,7 +428,9 @@ impl DefPathData {
             ClosureExpr => DefPathDataName::Anon { namespace: sym::closure },
             Ctor => DefPathDataName::Anon { namespace: sym::constructor },
             AnonConst => DefPathDataName::Anon { namespace: sym::constant },
-            ImplTrait | ImplTraitAssocTy => DefPathDataName::Anon { namespace: sym::opaque },
+            ImplTrait | ImplTraitAssocTy | AnonStruct | AnonUnion => {
+                DefPathDataName::Anon { namespace: sym::opaque }
```

Comment:
Could you change the name here?
`opaque` really means `impl Trait` (in HIR we have OpaqueTy and OpaqueDef).